tsconfig.json

```diff
@@ -40,6 +40,9 @@
     // Disallow inconsistently-cased references to the same file.
     "forceConsistentCasingInFileNames": true,
 
+    // Keep type of the variable in a catch clause as `any`
+    "useUnknownInCatchVariables": false,
```

Comment:
Why do we want to keep it as any? Is there a significant downside of using `unknown`? Will it involve a substantial amount of extra work? Could it be done in a followup task?

Reply:
Typescript can enhance code quality, but it is also important to keep balance between type safety and ease of use

The `useUnknownInCatchVariables` option is useful because it enforces a higher level of type safety in `catch` blocks. However, enabling this option can also lead to additional complexity in code. We would need to add extra type checking code in each `catch` block.

Let me show on examples

It is common to expect catch variable as instance of Error:
``` typescript
try {
  throw Error('error message');
} catch (error) {
  console.log(error.message); // => LOG: 'error message'
}
```

but it is not always the case:
``` typescript
try {
  throw 'error message';
} catch (error) {
  console.log(error.message); // => LOG: undefined
}
```

`useUnknownInCatchVariables` is intended to catch errors like this:
``` typescript
try {
  throw 123;
} catch (error) {
  if (err instanceof Error) {
    console.log(err.message);
  } else if (typeof err === 'string') {
    console.log(err);
  } else {
    console.log('Unexpected error')

  }
}
```

or:
``` typescript
try {
  throw 123;
} catch (error) {
  const e = error instanceof Error ? error : Error( `Unexpected Error: ${error}`);
   console.log(e.message);
}
```

Same time it is much easier to avoid checks by ignoring types:
``` typescript
try {
    throw new OwnError(123);
} catch (error: unknown) {
    console.log((error as OwnError).message);
}
```

or even: 
``` typescript
try {
  throw new Error(123);
} catch (error : any) {
   console.log(error.message);
}
```


While enabling `useUnknownInCatchVariables` could potentially catch errors, it might also introduce extra complexity into codebase. I am cautious about making this change, especially since the `useUnknownInCatchVariables` is enabled by default.

 I would be happy to make this change, if benefits of enabling `useUnknownInCatchVariables` outweigh potential downsides. Please let me know if you want to proceed with this and I will create a followup task

Reply:
It's a good idea to reduce complexity and remove conditions inside the catch block, but I think type safety and robust error handling have a higher weight than reducing complexity like that, so for me, it's better to use `unknown` rather than `any` for error variables in catch clauses

Reply:
There seems to be some confusion regarding this change. I have created issue #887 to do refactoring with `useUnknownInCatchVariables ` in separate task. It will help to keep this pull request limited. 

Could you please review it @BSFishy  @SergeyMyssak ?

Reply:
I'm wondering if we can avoid this change? I see that you have created an issue to remove it

Reply:
`useUnknownInCatchVariables` is not in tsconfig anymore. We have changes in this pull request to close #887